dbms/src/Common/OptimizedRegularExpression.h

```diff
@@ -58,6 +58,26 @@ struct Match
 };
 } // namespace OptimizedRegularExpressionDetails
 
+struct Instruction
+{
+    explicit Instruction(int substitution_num_)
+        : substitution_num(substitution_num_)
+    {}
+    explicit Instruction(String literal_)
```

Comment:
String && ？

Reply:
> String && ？

The value passed into this constructor will be repetedly used, I think this will break the `move`'s semantic.